### PR TITLE
(CDAP-1741) Docs for app logbacks

### PR DESCRIPTION
**Pull request by ParthGandhi**

### CODE REVIEW

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _application-logback:
+
+=====================
+Application Logbacks
+=====================
+
+YARN containers spawned by a CDAP application use the default container logback, `logback-container.xml`,
+packaged with CDAP. This `logback.xml` does log rotation and depending on the use case, may be sufficient.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: does log rotation and, depending on the use case, ... \[add comma]

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+However, it is also possible to specify a `logback.xml` for a CDAP application.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]:
```
However,  it is also possible to specify a custom ``logback.xml`` for a CDAP application.
```

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+The packaged `logback.xml` is then used for each container spawned by a program of the application.
+
+Writing and packaging a logback.xml
+===================================
+To specify a `logback.xml` to be used by the application, it is sufficient to package the file as a resource in the jar.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: To specify a `logback.xml` for use by an application, all that's required is to package the XML file as a resource in the application jar file.

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _application-logback:
+
+=====================
+Application Logbacks
+=====================
+
+YARN containers spawned by a CDAP application use the default container logback, `logback-container.xml`,
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: Have you been writing Markdown? Two back-ticks (fore and aft) are needed for a literal in reST.

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+packaged with CDAP. This `logback.xml` does log rotation and depending on the use case, may be sufficient.
+However, it is also possible to specify a `logback.xml` for a CDAP application.
+The packaged `logback.xml` is then used for each container spawned by a program of the application.
+
+Writing and packaging a logback.xml
+===================================
+To specify a `logback.xml` to be used by the application, it is sufficient to package the file as a resource in the jar.
+The `pom.xml` of the application may look like this, with the `resources` directory containing the `logback.xml`::
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: The `pom.xml` of the application, with the application's `resources` directory containing the `logback.xml`, might look like this::

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            <directory>resources</directory>
+          </resource>
+        </resources>
+      </build>
+
+    ...
+
+  </project>
+
+CDAP supports standard logback functionality. To write a custom logback you may
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: To write a custom logback,  you may

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+refer to `LOGBack <http://logback.qos.ch/>`__. An important thing to note is that when a `logback.xml` is
+specified for an application, CDAP does not handle log rotation. It is usually a good idea for your logback
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: CDAP then does not handle log rotation.

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+          </resource>
+        </resources>
+      </build>
+
+    ...
+
+  </project>
+
+CDAP supports standard logback functionality. To write a custom logback you may
+refer to `LOGBack <http://logback.qos.ch/>`__. An important thing to note is that when a `logback.xml` is
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: It's important to note that when a ``logback.xml`` is

File: cdap-docs/developers-manual/source/advanced/application-logback.rst [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+specified for an application, CDAP does not handle log rotation. It is usually a good idea for your logback
+to do rotation and expiry, to ensure that long running containers don't fill up the disk.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: long-running

**Comment by awholegunch** [comment]
A few comments. I can look at it again.

File: cdap-docs/developers-manual/source/advanced/application-logback.rst (hunk 4 above)
Comment (ParthGandhi) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: Ah of course. Stupid me.

File: cdap-docs/developers-manual/source/advanced/application-logback.rst (hunk 5 above)
Comment (ParthGandhi) [inline cdap-docs/developers-manual/source/advanced/application-logback.rst]: Yes, much better.

**Comment by ParthGandhi** [comment]
Re-opened against release/2.8: https://github.com/caskdata/cdap/pull/1607

**Pull request closed without merging**